src/GafferSceneUIModule/SetEditorBinding.cpp

```diff
@@ -407,6 +407,422 @@ class SetNameColumn : public StandardPathColumn
+			const auto includedSetMembers = setMembers->readable().intersection( visibleSet.inclusions );
+			result.value = new IntData( includedSetMembers.size() );
+			if( includedSetMembers.isEmpty() )
+			{
+				return result;
+			}
+
+			size_t excludedSetMemberCount = 0;
+			if( !visibleSet.exclusions.isEmpty() )
+			{
+				for( IECore::PathMatcher::Iterator it = includedSetMembers.begin(), eIt = includedSetMembers.end(); it != eIt; ++it )
+				{
+					const auto visibility = visibleSet.visibility( *it );
+					if( visibility.drawMode != GafferScene::VisibleSet::Visibility::Visible )
+					{
+						excludedSetMemberCount++;
+					}
+				}
+			}
+
+			iconData->writable()["state:highlighted"] = g_setIncludedHighlightedIconName;
+			const bool allSetMembersIncluded = includedSetMembers.size() == setMembers->readable().size();
+			if( excludedSetMemberCount == 0 )
+			{
+				iconData->writable()["state:normal"] = allSetMembersIncluded ? g_setIncludedIconName : g_setPartiallyIncludedIconName;
+				result.toolTip = allSetMembersIncluded ? g_setIncludedToolTip : g_setPartiallyIncludedToolTip;
+			}
+			else if( includedSetMembers.size() == excludedSetMemberCount )
+			{
+				iconData->writable()["state:normal"] = g_setIncludedDisabledIconName;
+				result.toolTip = allSetMembersIncluded ? g_setIncludedOverrideToolTip : g_setPartiallyIncludedOverrideToolTip;
+			}
+			else
+			{
+				iconData->writable()["state:normal"] = g_setPartiallyDisabledIconName;
+				result.toolTip = allSetMembersIncluded ? g_setIncludedPartialOverrideToolTip : g_setPartiallyIncludedPartialOverrideToolTip;
+			}
+
+			return result;
+		}
+
+		CellData headerData( const IECore::Canceller *canceller ) const override
+		{
+			const auto visibleSet = ContextAlgo::getVisibleSet( m_context.get() );
+			return CellData( /* value = */ nullptr, /* icon = */ visibleSet.inclusions.isEmpty() ? g_inclusionsEmptyIconName : g_setIncludedIconName, /* background = */ nullptr, /* tooltip = */ new StringData( "Visible Set Inclusions" ) );
+		}
+
+	private :
+
+		void contextChanged( const IECore::InternedString &name )
+		{
+			if( ContextAlgo::affectsVisibleSet( name ) )
+			{
+				changedSignal()( this );
+			}
+		}
+
+		bool buttonPress( const ButtonEvent &event )
+		{
+			if( event.buttons != ButtonEvent::Left )
+			{
+				return false;
+			}
+
+			return true;
+		}
+
+		bool buttonRelease( const Gaffer::Path &path, const GafferUI::PathListingWidget &widget, const ButtonEvent &event )
+		{
+			auto setPath = IECore::runTimeCast<const SetPath>( &path );
+			if( !setPath )
+			{
+				return false;
+			}
+
+			const auto setName = runTimeCast<const IECore::StringData>( setPath->property( g_setNamePropertyName ) );
+			if( !setName )
+			{
+				// We only interact with locations representing sets
+				return false;
+			}
+
+			auto paths = IECore::PathMatcher();
+			const auto setMembers = setPath->getScene()->set( setName->readable() );
+			paths.addPaths( setMembers->readable() );
```

Comment:
I think this is all equivalent to this one liner : `auto paths = setPath->getScene()->set( setName->readable() ).readable()`.

Reply:
Think we still need `setMembers` separately in order to determine the toggle behaviour at [L566](https://github.com/GafferHQ/gaffer/blob/32e50f98a2c90fe65824ca08356181c78cbd5368/src/GafferSceneUIModule/SetEditorBinding.cpp#L566), but I've adjusted this a bit in e873517.